src/YesSql.Provider.InMemory/InMemoryDbProviderOptionsExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.Data.Sqlite;
+using System;
+using System.Data;
+using YesSql.Core.Provider;
+using YesSql.Core.Services;
+using YesSql.Storage.InMemory;
+
+namespace YesSql.Provider.InMemory
+{
+    public static class InMemoryDbProviderOptionsExtensions
+    {
+        private const string ConnectionString = "Data Source=:memory:";
+
+        public static void UseInMemory(
+            this IDbProviderOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            var configuration = new Configuration
+            {
+                ConnectionFactory = new DbConnectionFactory<SqliteConnection>(ConnectionString, true),
```

Comment:
Interesting idea, use the in-memory storage with an in-memory sqllite index. Didn't think about it.